src/Microsoft.Health.Fhir.Api/Features/Filters/OperationOutcomeExceptionFilterAttribute.cs

```diff
@@ -66,6 +66,7 @@ public override void OnActionExecuted(ActionExecutedContext context)
 
                         break;
                     case ResourceNotFoundException _:
+                    case JobNotFoundException _:
```

Comment:
In case of JobNotFoundException, shouldn't it return OperationResult instead of FhirResult?

Reply:
Oh I see what it's doing. It's using the existing code path of FhirResult to serialize the result into OperationOutcome. Might still need some thinking around this one since in one place we are separating the job stuff into OperationResult but here it's tight back to FhirResult.

Reply:
Added a new ActionResult that will be used for handling all these failure cases. Now ExportController will return a OperationResult, FhirController a FhirResult and the OperationOutcomeFilterAttribute will return a OperationOutcomeResult